bson-record-codec/src/main/org/bson/codecs/record/RecordCodec.java

```diff
@@ -218,16 +254,47 @@ private static <T extends Annotation> void validateAnnotationOnlyOnField(final R
 
     RecordCodec(final Class<T> clazz, final CodecRegistry codecRegistry) {
         this.clazz = notNull("class", clazz);
+        if (clazz.getTypeParameters().length > 0) {
```

Comment:
In this state, the codec is unusable unless it is first parameterized.  That effectively means that we don't support encoding records where the outermost record has type parameters.  This is the same restriction as we have for POJOs.